### PR TITLE
switch docker files based on cpu arch

### PR DESCRIPTION
**Pull request by vyasworks**
## Objective 

Enable M1 users to use same commands as non M1 users.

### CODE REVIEW

File: docker-compose.M1.yaml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      interval: 5s
+      timeout: 5s
+      retries: 5
+
+  ####################### Flyway Migration Definition #######################
+  flyway:
+    container_name: mds_flyway
+    build:
+      context: migrations
+      dockerfile: Dockerfile.flyway.M1.dev
```
Comment (vyasworks) [inline docker-compose.M1.yaml:31]: This file is a copy of `docker-compose` Just this docker file is `M1`

File: docker-compose.M1.yaml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    healthcheck:
+      test: [ "CMD", "curl", "localhost:3000/health" ]
+      interval: 15s
+      timeout: 5s
+      retries: 5
+      start_period: 30s
+
+  ####################### Minespace Definition #######################
+  minespace:
+    restart: always
```
Comment (hitankar-freshworks) [inline docker-compose.M1.yaml:116]: `platform: linux/amd64`

File: docker-compose.M1.yaml [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    healthcheck:
+      test: [ "CMD", "redis-cli", "ping" ]
+      interval: 5s
+      timeout: 5s
+      retries: 5
+
+  ####################### Frontend Definition #######################
+  frontend:
+    restart: always
+    user: 1000:1000
```
Comment (hitankar-freshworks) [inline docker-compose.M1.yaml:94]: `platform: linux/amd64`

Comment (vyasworks) [reply docker-compose.M1.yaml:94]: done

Comment (vyasworks) [reply docker-compose.M1.yaml:94]: done

**Review by hitankar-freshworks — APPROVED** [review]

**Pull request merged**